File: src/Presentation/scripts/models/MatchListItem.ts
export interface MatchListItem {
    id: string,
    url: string
    name: string,
    password: string,
    numberOfConnections: number
}

export const getMatchFullUrl = function (matchId: string): string {
    const domain: string = window.location.host;
    const protocol: string = window.location.protocol;

    const o_fullUrl: string = `${protocol}//${domain}/game/${matchId}`;
    return o_fullUrl
};

File: src/Application/Common/Models/MatchListItem.cs
namespace Application.Common.Models
{
    public class MatchListItem
    {
        public string Id { get; set; }

        public string Url { get; set; }

        private string _name;
        public string Name
        {
            get => _name;
            set
            {
                if(value.Length > 85)
                {
                    _name = value.Substring(0, 82) + "...";
                    return;
                }

                _name = value;
            }
        }

        public string Password { get; set; }

        public int NumberOfConnections { get; set; }
    }
}
